src/chapter4/challenges.md

```diff
@@ -4,48 +4,57 @@
 
 ## Task 1 - Parallise `for` Loop
 
-Goal: To to create an array `[0,1,2...19]`
+Goal: To to create an array `[0,1,2...100000]`
 
 1. Git clone [HPC-Training-Challenges](https://github.com/MonashDeepNeuron/HPC-Training-Challenges)
-2. Go to the directory “challenges/parallel-computing”. Compile array.c and execute it. Check the run time of the serial code
-3. Add `#pragma<>` 
-4. Compile the code again
-5. Run parallel code and check the improved run time
+2. Go to the directory “challenges/parallel-computing” and open `array.c` file
+3. Implement the code to create an array `[0,1,2...100000]` without parallelisation
+4. Measure the run time of the code
+5. Use `#pragma<>` and potentially other clauses to parallelise the code
+6. Compile the code again and check the run time and observe the result
 
 ## Task 2 - Run task 1 on HPC cluster
 
-1.  Check the available partitions with `show_cluster`
-2.  Modify `RunHello.sh `
-3.  `sbatch RunHello.sh`
-4.  `cat slurm<>.out` and check the run time
+1. Log into M3
+2. Check the available partitions with `show_cluster`
+3. Modify `RunHello.sh` to you can run `array.c` on HPC cluster
+4. Submit the job to M3
+5. Check the slurm output file
 
 >You can also use [strudel web](https://beta.desktop.cvl.org.au/login) to run the script without sbatch
 
 ## Task 3 - Reduction Clause
 
 Goal: To find the sum of the array elements
 
-1.  Compile `reduction.c` and execute it. Check the run time
-2.  Add `#pragma<>`
-3.  Compile `reduction.c` again
-4.  Run parallel code and check the improved run time. Make sure you got the same result as the serial code
+1. Implement the code in `reduction.c` to find the sum of the array elements without parallelisation 
+2. Measure the run time of the code
+3. Add `#pragma<>` and potentially other clauses to parallelise the code
+4. Compile and run `reduction.c` again
+5. Check the run time and observe the result 
 
 >`module load gcc` to use newer version of gcc if you have error with something like `-std=c99`
 
 ## Task 4 - Private clause
 
 The goal of this task is to square each value in array and find the sum of them
-1.  Compile private.c and execute it. Check the run time. `#include` the default library `<math.h>` and link it
-2.  Add `#pragma<>`
-3.  Compile `private.c` again
-4.  Run parallel code and check the improved run time
+
+1.  Implement the code in `private.c` to square each value in array and find the sum of them without parallelisation
```

Comment:
Maybe rename the `private.c` file to `power.c`, `map.c` or `transform.c` (latter two being generic names for the algorithms similar to how reduce is sometimes called fold) if raising a number to a power is all the program is doing.

Reply:
Renamed to sumPowers to be more descriptive of the function of the program

Reply:
Thanks, in future can we keep a snake or kebab case for filenames (eg. `sum-of-powers.c`). I'll make sure the CONTRIBUTING doc is up-to-date for the challenges repo.